test/widget/crowd_league_app/crowd_league_app_test.dart

```diff
@@ -0,0 +1,93 @@
+import 'dart:async';
+
+import 'package:crowdleague/actions/auth/observe_auth_state.dart';
+import 'package:crowdleague/actions/database/plumb_database_stream.dart';
+import 'package:crowdleague/actions/device/check_platform.dart';
+import 'package:crowdleague/actions/notifications/print_fcm_token.dart';
+import 'package:crowdleague/actions/notifications/request_fcm_permissions.dart';
+import 'package:crowdleague/models/app/app_state.dart';
+import 'package:crowdleague/reducers/app_reducer.dart';
+import 'package:crowdleague/widgets/crowd_league_app.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:redux/redux.dart';
+
+import '../../mocks/wrappers/firebase_wrapper_mocks.dart';
+import '../../utils/services_bundle_mocks.dart';
+import '../../utils/verify_dispatch_middleware.dart';
+
+void main() {
+  group('CrowddLeagueApp', () {
+    testWidgets('shows expected UI while initializing',
+        (WidgetTester tester) async {
+      // create a fake(ish) a services bundle
+      final reduxCompleter = Completer<Store<AppState>>();
+      final redux = FakeServicesBundle(completer: reduxCompleter);
+
+      // create a fake firebase wrapper with a supplied completer
+      final firebaseCompleter = Completer<FirebaseApp>();
+      final firebase = FakeFirebaseWrapper(completer: firebaseCompleter);
+
+      // the widget under test
+      final wut = CrowdLeagueApp(redux: redux, firebase: firebase);
+
+      // Tell the tester to build the widget tree.
+      await tester.pumpWidget(wut);
+
+      // Create the Finders.
+      final initializingIndicatorFinder = find.byType(InitializingIndicator);
+
+      // verify that the InitializingIndicator is shown
+      expect(initializingIndicatorFinder, findsOneWidget);
+
+      // verify the expected text is shown, indicating waiting for both
+      expect(find.text('Waiting for Firebase...'), findsOneWidget);
+
+      // complete the firebase future and verfiy text has changed as expected
+      firebaseCompleter.complete();
```

Comment:
all the setup was well worth it. thanks for pushing through!!

Reply:
Yeah good to have figured it out I reckon, should be real helpful down the track I'd say

Reply:
Thanks mate!